src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -196,6 +196,11 @@ export class ConnectionDialogService implements IConnectionDialogService {
 					profile.providerName = Constants.mssqlProviderName;
 				}
 
+				// Determine if database has been specified to connect to
+				if (profile.databaseName && !(profile.databaseName === `''`)) {
```

Comment:
why do we need to check for \`''\`? this can be simplified to if(profile.databaseName)

Reply:
I have seen when database is not selected in connection dialog it shows that databaseName as '' vs undefined.

Reply:
Uhhh, that seems like a bug...it's definitely not a valid DB name so we shouldn't have that value there.